app/src/main/java/com/example/cachua/musicdemo21/service/MyMusicService.kt

```diff
@@ -0,0 +1,242 @@
+                playSong()
+                mCallBack.onUpTime(mMusic, mMediaPlayer.currentPosition)
+            }
+        } else {
+            if (intent?.action.equals(Constant.NEXT_ACTION)) {
+                nextSong()
+            } else {
+                if (intent?.action.equals(Constant.PREV_ACTION)) {
+                    previousSong()
+                }
+            }
+        }
+        return START_NOT_STICKY
+    }
+
+
+    override fun onBind(intent: Intent): IBinder {
+        return mIbinder
+    }
+
+    override fun onUnbind(intent: Intent?): Boolean {
+        return super.onUnbind(intent)
+    }
+
+    override fun onDestroy() {
+        mMediaPlayer.stop()
+        super.onDestroy()
+    }
+
+
+    inner class playMusicBinder : Binder() {
+        fun service(): MyMusicService {
+            return this@MyMusicService
+        }
+    }
+
+
+    fun isPlaying(): Int {
+        return mStatusService
+    }
+
+    fun onProgress(i: Int) {
+        mMediaPlayer.seekTo(i * mMusic.duration / 100)
+    }
+
+
+    fun selectSong(mContext: Context, mArrayListMusic: ArrayList<MusicModel>, position: Int) {
+        this.mArrayListMusic = mArrayListMusic
+        mPosition = position
+        mMediaPlayer.reset()
+        mMediaPlayer.setDataSource(mContext, Uri.parse(mArrayListMusic[position].data))
+        mMediaPlayer.prepare()
+        mMusic = mArrayListMusic[position]
+
+        if (mStatusService == 0) {
+            mMediaPlayer.start()
+        }
+        upDateMusicTime(mMusic, 0, mPosition)
+    }
+
+    fun nextSong() {
+        mPosition = if (mPosition < mArrayListMusic.size - 1) {
+            mPosition + 1
+        } else {
+            0
+        }
+        mMediaPlayer.reset()
+        mMediaPlayer.setDataSource(this, Uri.parse(mArrayListMusic[mPosition].data))
+        mMediaPlayer.prepare()
+        mMusic = mArrayListMusic[mPosition]
+        if (mStatusService == 0) {
+            mMediaPlayer.start()
+        }
+        upDateMusicTime(mMusic, 0, mPosition)
+    }
+
+    fun previousSong() {
+        mPosition = if (mPosition <= 0) {
+            mArrayListMusic.size - 1
+        } else {
+            mPosition - 1
+        }
+        mMediaPlayer.reset()
+        mMediaPlayer.setDataSource(this, Uri.parse(mArrayListMusic[mPosition].data))
+        mMediaPlayer.prepare()
+        mMusic = mArrayListMusic[mPosition]
+        if (mStatusService == 0) {
+            mMediaPlayer.start()
+        }
+        upDateMusicTime(mMusic, 0, mPosition)
+
+    }
+
+    fun playSong() {
+        mMediaPlayer.start()
+        mStatusService = 0
+        upDateMusicTime(mMusic, mMediaPlayer.currentPosition, mPosition)
+        pushNotification(mMusic)
+    }
+
+    fun pauseSong() {
+        mMediaPlayer.pause()
+        mStatusService = 1
+        pushNotification(mMusic)
+    }
+
+    fun upDateMusicTime(music: MusicModel, progress: Int, position: Int) {
+        Handler().postDelayed({
+            if (progress == 0) {
```

Comment:
Với những câu lệnh nào if else có thể trên 1 dòng thì không cần {}, các trường hợp khác cần có {}
```suggestion
            if (progress == 0) pushNotification(music)
```